doc/man3/OPENSSL_init_crypto.pod

```diff
@@ -190,15 +190,17 @@ OPENSSL_init_crypto(). For example:
 
 The OPENSSL_cleanup() function deinitialises OpenSSL (both libcrypto
 and libssl). All resources allocated by OpenSSL are freed. Typically there
-should be no need to call this function directly as it is initiated
-automatically on application exit. This is done via the standard C library
-atexit() function. In the event that the application will close in a manner
-that will not call the registered atexit() handlers then the application should
-call OPENSSL_cleanup() directly. Developers of libraries using OpenSSL
-are discouraged from calling this function and should instead, typically, rely
-on auto-deinitialisation. This is to avoid error conditions where both an
-application and a library it depends on both use OpenSSL, and the library
-deinitialises it before the application has finished using it.
+should be no need to call this function directly as it is done 
+automatically on application exit. This is arranged by linker which ensures
+library destructor gets invoked with library unload (or process exit).
+On platforms where linker can not arrange invocation of library destructor,
+the application may call OPENSSL_cleanup() to dispose all resources
+allocated by libcrypto/libssl.  Developers of libraries using OpenSSL
+are discouraged from calling OPENSSL_cleanup() function and should instead,
+rely on linker to invoke OPENSSL_cleanup() via library destructor.
+Explicit call to OPENSSL_cleanup() from 3rd party library may disable
+libcrypto/libssl for application prematurely as calls to those
+libraries typically fail after OPENSSL_cleanup() is done.
```

Comment:
```suggestion
should be no need to call this function directly as it is done 
automatically on application exit. This is arranged by the linker which ensures
a library destructor gets invoked on library unload (or process exit).
On some platforms there is no support for library destructors. On these platforms
the OS will still automatically dispose of all resources on process exit. However
memory leak detection tools such as valgrind or asan may report spurious memory
leaks. The application may call OPENSSL_cleanup() explicitly to dispose of all
resources allocated by libcrypto/libssl to avoid this.  Developers of libraries using OpenSSL
are discouraged from calling the OPENSSL_cleanup() function and should instead,
rely on the linker to invoke OPENSSL_cleanup() via the library destructor.
Explicit calls to OPENSSL_cleanup() from 3rd party libraries may disable
libcrypto/libssl for the application prematurely as calls to those
libraries typically fail after OPENSSL_cleanup() is done.
```

Reply:
I'm accepting version from @mattcaswell with minor tweak:
```
 On some platforms (or if your application is linked statically) there is no support
```